metric/instrument.go

```diff
@@ -171,6 +177,23 @@ func (o unitOpt) applyInt64ObservableGauge(c Int64ObservableGaugeConfig) Int64Ob
 // The unit u should be defined using the appropriate [UCUM](https://ucum.org) case-sensitive code.
 func WithUnit(u string) InstrumentOption { return unitOpt(u) }
 
+// WithExplicitBucketBoundaries sets the instrument explicit bucket boundaries.
+//
+// This option is considered "advisory", and may be ignored by API implementations.
+func WithExplicitBucketBoundaries(bounds ...float64) HistogramOption { return bucketOpt(bounds) }
+
+type bucketOpt []float64
+
+func (o bucketOpt) applyFloat64Histogram(c Float64HistogramConfig) Float64HistogramConfig {
+	c.explicitBucketBoundaries = o
```

Comment:
Just want to double-check, the SDK does check if the bounds are empty, eg. [], it sets to something sane and it doesn't have to be handled here.

Reply:
The [API spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-advisory-parameter-explicitbucketboundaries) doesn't say anything about validation.  In my PR implementing this in the sdk, I added validation:

https://github.com/open-telemetry/opentelemetry-go/pull/4605/files#diff-e6ec718e2bb8f9b29cbab6b14b62ddae15faa6192d82b927b0e7b7f1b4167cc7R503